packages/store/tests/issues/canceling-promises.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { Injectable } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { State, Action, Store, provideStore } from '@ngxs/store';
+
+import { createPromiseTestHelper } from '../helpers/promise-test-helper';
+
+// This test essentially shows that microtasks are not cancelable in JavaScript.
+// Therefore, using actions that return promises cannot be used with the
+// `cancelUncompleted` option, as this will result in the functionality being
+// executed twice (once by the executing action and again by the newly dispatched
+// action). Any third-party code that returns promises is also not cancelable.
+describe('Canceling promises', () => {
+  const recorder: string[] = [];
+
+  class MyActionAwait {
+    static readonly type = '[MyState] My action await';
+  }
+
+  class MyActionThen {
+    static readonly type = '[MyState] My action then';
+  }
+
+  const { promise: promiseAwaitReady, markPromiseResolved: markPromiseAwaitReady } =
+    createPromiseTestHelper();
+
+  const { promise: promiseThenReady, markPromiseResolved: markPromiseThenReady } =
+    createPromiseTestHelper();
+
+  @State<string>({
+    name: 'myState',
+    defaults: 'STATE_VALUE'
+  })
+  @Injectable()
+  class MyState {
+    @Action(MyActionAwait, { cancelUncompleted: true })
+    async handleActionAwait() {
+      recorder.push('before promise await ready');
+      await promiseAwaitReady;
+      recorder.push('after promise await ready');
+    }
+
+    @Action(MyActionThen, { cancelUncompleted: true })
+    handleActionThen() {
+      recorder.push('before promise then ready');
+      return promiseThenReady.then(() => {
+        recorder.push('after promise then ready');
+      });
```

Comment:
What are we testing here?
There isn't anything that checks that operations on the state are nooped after cancellation.